lib/elixir/lib/kernel.ex

```diff
@@ -795,11 +798,13 @@ defmodule Kernel do
 
       iex> trunc(5.4)
       5
+
       iex> trunc(5.99)
       5
 
   """
-  @spec trunc(number) :: integer
+  @spec trunc(arg) :: arg when arg: integer
```

Comment:
Mh, I see. But to me it doesn't really make sense, for example, say you have a function that takes a function and returns the result of invoking the argument function, doubled:

```elixir
def call_then_double(fun) when is_function(fun, 0) do
  fun.() * 2
end
```

I would spec this as:

```elixir
@spec call_then_double((() -> result)) :: result when result: integer | float
```

but this doesn't mean that since the argument fun returns `result`, then the outer fun returns `result`, it rather means (at least in my head 😃) that if the argument function returns a value of type `result`  then `call_then_double/1` will return a value of the same type. 😕